packages/nextjs/src/index.server.ts

```diff
@@ -111,6 +111,12 @@ function addServerIntegrations(options: NextjsOptions): void {
   });
   integrations = addOrUpdateIntegration(defaultRewriteFramesIntegration, integrations);
 
+  const defaultRequestDataIntegration = new RequestData({ _addReqDataCallback: addRequestDataToEvent });
+  integrations = addOrUpdateIntegration(defaultRequestDataIntegration, integrations, {
+    // Specify the `@sentry/node` version of `addRequestDataToEvent`, so we get the injected dependencies
+    '_options._addReqDataCallback': addRequestDataToEvent,
+  });
```

Comment:
Rant alert. Just want to get the following out for my own sanity - no ill intentions:

I don't have any actionable feedback here but these lines of code make me question whether it is a good idea for us to internally use integrations at all. This whole `addOrUpdateBlah` api + passing down integrations and having to deal with them competing with eachother is so weird to me. I feel like we're more often fighting integrations than using them. In my perfect world, anything that is default should just be init options. Only extra non-default functionality should be done via integrations.

Reply:
Yeah, this `addOrUpdate` business is a little weird, I'll admit, though it predates me. I actually have most of a solution to at least make that part cleaner, but it would mean making core code have more bytes just to accommodate something which only the nextjs SDK needs, so I gave up on it.

As for the rest, I wasn't there when the unified SDK API was specced out, but I think the reasoning is something along the lines of "keeping SDK-specific default behavior in integrations allows the init options to be more standardized across SDKs (which is the ultimate goal of the the unified API)."